core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -218,7 +218,13 @@ protected ExprEval eval(long param)
 
     protected ExprEval eval(double param)
     {
-      return eval((long) param);
+      long param2;
+      if (param > Long.MIN_VALUE && param < Long.MAX_VALUE) {
+        param2 = (long) param;
```

Comment:
This `param2` variable seems unnecessary, you can directly use `return eval((long) param)` instead.

Reply:
thanks for pointing, have removed unnecessary variable

Reply:
@isandeep41 Cool, thanks